common/src/main/java/generations/gg/generations/core/generationscore/client/model/GenericObjectPool.java

```diff
@@ -0,0 +1,33 @@
+package generations.gg.generations.core.generationscore.client.model;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.function.Supplier;
+
+public class GenericObjectPool<T> {
+    private final Queue<T> availableObjects;
+    private final Queue<T> usedObjects;
+
+    public GenericObjectPool(Supplier<T> objectFactory, int maxPoolSize) {
+        this.availableObjects = new ArrayDeque<>(maxPoolSize);
+        this.usedObjects = new ArrayDeque<>(maxPoolSize);
+        for (int i = 0; i < maxPoolSize; i++) {
+            availableObjects.add(objectFactory.get());
```

Comment:
should be able to copy all over using List method but not sure since supplier is funny